javascripts/discourse/components/welcome-link-banner.js

```diff
@@ -1,40 +1,61 @@
 import Component from "@ember/component";
-import discourseComputed from "discourse-common/utils/decorators";
+import { action } from "@ember/object";
 import { inject as service } from "@ember/service";
-import { computed } from "@ember/object";
 import { defaultHomepage } from "discourse/lib/utilities";
-export default Component.extend({
-  router: service(),
+import { tracked } from "@glimmer/tracking";
 
-  bannerLinks: computed(function () {
-    return JSON.parse(settings.banner_links);
-  }),
+export default class WelcomeLinkBanner extends Component {
+  @service router;
+  @service currentUser;
```

Comment:
Since you're using `this.site`:

```suggestion
  @service currentUser;
  @service site;

```